src/NodeApi/Runtime/NodeEmbeddingRuntime.cs

```diff
@@ -0,0 +1,161 @@
+            .ThrowIfFailed();
+        return result;
+    }
+
+    public static unsafe void Run(
+        NodeEmbeddingPlatform platform, NodeEmbeddingRuntimeSettings? settings = null)
+    {
+        using FunctorRef<node_embedding_runtime_configure_callback> functorRef =
+            CreateRuntimeConfigureFunctorRef(settings?.CreateConfigureRuntimeCallback());
+        JSRuntime.EmbeddingRunRuntime(platform.Handle, functorRef.Callback, functorRef.Data)
+            .ThrowIfFailed();
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether the Node.js environment is disposed.
+    /// </summary>
+    public bool IsDisposed { get; private set; }
+
+    /// <summary>
+    /// Disposes the Node.js environment, causing its main thread to exit.
+    /// </summary>
+    public void Dispose()
+    {
+        if (IsDisposed || !_mustDeleteRuntime) return;
+        IsDisposed = true;
+
+        JSRuntime.EmbeddingDeleteRuntime(Handle).ThrowIfFailed();
+    }
+
+    public unsafe void RunEventLoop()
+    {
+        if (IsDisposed) throw new ObjectDisposedException(nameof(NodeEmbeddingRuntime));
+
+        JSRuntime.EmbeddingRuntimeRunEventLoop(Handle).ThrowIfFailed();
+    }
+
+    public unsafe void TerminateEventLoop()
+    {
+        if (IsDisposed) throw new ObjectDisposedException(nameof(NodeEmbeddingRuntime));
+
+        JSRuntime.EmbeddingRuntimeTerminateEventLoop(Handle).ThrowIfFailed();
+    }
+
+    public unsafe bool RunEventLoopOnce()
+    {
+        if (IsDisposed) throw new ObjectDisposedException(nameof(NodeEmbeddingRuntime));
+
+        JSRuntime.EmbeddingRuntimeRunOnceEventLoop(Handle, out bool result).ThrowIfFailed();
+        return result;
+    }
+
+    public unsafe bool RunEventLoopNoWait()
+    {
+        if (IsDisposed) throw new ObjectDisposedException(nameof(NodeEmbeddingRuntime));
+
+        JSRuntime.EmbeddingRuntimeRunNoWaitEventLoop(Handle, out bool result).ThrowIfFailed();
+        return result;
+    }
+
+    public unsafe void RunNodeApi(RunNodeApiCallback runNodeApi)
+    {
+        if (IsDisposed) throw new ObjectDisposedException(nameof(NodeEmbeddingRuntime));
+
+        using FunctorRef<node_embedding_node_api_run_callback> functorRef =
+            CreateNodeApiRunFunctorRef(runNodeApi);
+        JSRuntime.EmbeddingRuntimeRunNodeApi(Handle, functorRef.Callback, functorRef.Data)
+            .ThrowIfFailed();
+    }
+
+#if UNMANAGED_DELEGATES
+    private static readonly unsafe delegate* unmanaged[Cdecl]<nint, NodeEmbeddingStatus>
+        s_releaseRuntimeCallback = &ReleaseRuntimeCallbackAdapter;
+#else
+    private static readonly node_embedding_data_release_callback.Delegate
+        s_releaseRuntimeCallback = ReleaseRuntimeCallbackAdapter;
+#endif
+
+#if UNMANAGED_DELEGATES
+    [UnmanagedCallersOnly(CallConvs = new[] { typeof(CallConvCdecl) })]
+#endif
+    private static unsafe NodeEmbeddingStatus ReleaseRuntimeCallbackAdapter(nint data)
```

Comment:
This method is not used.

Reply:
It is used for initializing the `s_releaseRuntimeCallback` variable above which in turn used for releasing the user data in the code above:
```C#
        NodeEmbedding.JSRuntime.EmbeddingRuntimeSetUserData(
            runtime,
            (nint)GCHandle.Alloc(result),
            new node_embedding_data_release_callback(s_releaseRuntimeCallback))
```
The main point is to invalidate the wrapped `node_embedding_runtime` handler as soon as it is deleted by the LibNode.

In all other cases the `node_embedding_data_release_callback` is used just to release the `GCHandle`, but in this scenario we must run some `NodeEmbeddingRuntime` specific code.